### PR TITLE
feat: add updateArgo endpoint to change source repo, path, and label values

### PR DESCRIPTION
**Pull request by aharken**
- Adds new route to argocd backend plugin
  - `/updateArgo/:argoAppName`
  - Provides the ability to change the `sourceRepo`, `sourcePath`, and `labelValue` of an argo app and project
#### :heavy_check_mark: Checklist

- [x] Added tests for new functionality and regression tests for bug fixes
- [x] Added changeset (run `yarn changeset` in the root)
- [ ] Screenshots of before and after attached (for UI changes)
- [ ] Added or updated documentation (if applicable)

### CODE REVIEW

File: plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts [hunk 1]
```diff
@@ -639,4 +798,103 @@ export class ArgoService implements ArgoServiceApi {
 
     return true;
```
Comment (jpdeburgo) [inline plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts:799]: Why'd you add this here? Not against it. Just wondering

File: plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts [hunk 2]
```diff
@@ -198,12 +198,67 @@ export function createRouter({
...
+      console.log(e);
       return response.status(500).send({
         status: 500,
         message: e.message || 'Failed to create argo app',
       });
     }
   });
+  router.put('/updateArgo/:argoAppName', async (request, response) => {
+    const argoInstanceName: string = request.body.clusterName;
+    const namespace = request.body.namespace;
```
Comment (jpdeburgo) [inline plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts:209]:
```suggestion
    const namespace = request.body.namespace;
```

File: plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts [hunk 3]
```diff
@@ -198,12 +198,67 @@ export function createRouter({
...
+    const projectName = request.body.projectName as string;
+    const appName = request.body.appName as string;
+    const labelValue = request.body.labelValue as string;
+    const sourceRepo = request.body.sourceRepo;
```
Comment (jpdeburgo) [inline plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts:213]: const {sourceRepo} = request.body;
```suggestion
    const sourceRepo = request.body.sourceRepo;
```

File: plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts [hunk 4]
```diff
@@ -198,12 +198,67 @@ export function createRouter({
...
+    const sourcePath = request.body.sourcePath;
```
Comment (jpdeburgo) [inline plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts:214]:
```suggestion
    const sourcePath = request.body.sourcePath;
```

**Comment by jpdeburgo** [comment]
Actually they do that everywhere else, so it's chill

**Review by jpdeburgo — APPROVED** [review]
Very solid

**Review by karlhaworth — APPROVED** [review]
lgtm

**Pull request merged**